pkg/controllers/awsloadbalancercontroller/deployment.go

```diff
@@ -46,22 +52,40 @@ const (
 	boundSATokenVolumeName = "bound-sa-token"
 	// boundSATokenDir is the sa token directory
 	boundSATokenDir = "/var/run/secrets/openshift/serviceaccount"
+	// trustedCAVolumeName is the name of the volume with the CA bundle to be trusted by the controller.
+	trustedCAVolumeName = "trusted-ca"
+	// trustedCAPath is the mounting path for the trusted CA bundle.
+	// Default certificate path is taken from the golang source:
+	// https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/crypto/x509/root_linux.go;drc=82f09b75ca181a6be0e594e1917e4d3d91934b27;l=20
+	trustedCAPath = "/etc/pki/tls/certs/albo-tls-ca-bundle.crt"
+	// defaultCABundleKey is the default name for the data key of the configmap injected with the trusted CA.
+	defaultCABundleKey = "ca-bundle.crt"
 	// all capabilities in the pod security context
 	allCapabilities = "ALL"
 )
 
-func (r *AWSLoadBalancerControllerReconciler) ensureDeployment(ctx context.Context, namespace, image string, sa *corev1.ServiceAccount, crSecretName, servingSecretName string, controller *albo.AWSLoadBalancerController) (*appsv1.Deployment, error) {
+func (r *AWSLoadBalancerControllerReconciler) ensureDeployment(ctx context.Context, sa *corev1.ServiceAccount, crSecretName, servingSecretName string, controller *albo.AWSLoadBalancerController, trustCAConfigMap *corev1.ConfigMap) (*appsv1.Deployment, error) {
 	deploymentName := fmt.Sprintf("%s-%s", controllerResourcePrefix, controller.Name)
 
 	reqLogger := log.FromContext(ctx).WithValues("deployment", deploymentName)
 	reqLogger.Info("ensuring deployment for aws-load-balancer-controller instance")
 
-	exists, current, err := r.currentDeployment(ctx, deploymentName, namespace)
+	trustCAConfigMapName, trustCAConfigMapHash := "", ""
+	if trustCAConfigMap != nil {
+		trustCAConfigMapName = trustCAConfigMap.Name
+		configMapHash, err := buildMapHash(trustCAConfigMap.Data)
+		if err != nil {
+			return nil, fmt.Errorf("failed to build the trusted CA configmap's hash: %w", err)
+		}
+		trustCAConfigMapHash = configMapHash
+	}
```

Comment:
Move this section down to after L86.  In other words, check first to make sure the deployment exists before you do this work.

Reply:
Done.